includes/core/server/request/class-get-reporting-payment-activity.php

```diff
@@ -50,32 +49,48 @@ public function get_method(): string {
 	/**
 	 * Sets the start date for the payment activity data.
 	 *
-	 * @param string|null $date_start The start date in the format 'YYYY-MM-DDT00:00:00' or null.
+	 * @param string $date_start The start date in the format 'YYYY-MM-DDT00:00:00'.
 	 * @return void
 	 */
-	public function set_date_start( ?string $date_start ) {
-		// TBD - validation.
+	public function set_date_start( string $date_start ) {
+		$this->validate_date( $date_start, 'Y-m-d\TH:i:s' );
 		$this->set_param( 'date_start', $date_start );
 	}
 
 	/**
 	 * Sets the end date for the payment activity data.
 	 *
-	 * @param string|null $date_end The end date in the format 'YYYY-MM-DDT00:00:00' or null.
+	 * @param string $date_end The end date in the format 'YYYY-MM-DDT00:00:00'.
 	 * @return void
 	 */
 	public function set_date_end( string $date_end ) {
-		// TBD - validation.
+		$this->validate_date( $date_end, 'Y-m-d\TH:i:s' );
 		$this->set_param( 'date_end', $date_end );
 	}
 
 	/**
 	 * Sets the timezone for the reporting data.
 	 *
-	 * @param string|null $timezone The timezone to set or null.
+	 * @param string $timezone The timezone to set.
 	 * @return void
+	 *
+	 * @throws Invalid_Request_Parameter_Exception Exception if the timezone is not in valid format.
```

Comment:
Might add `@throws` to the PHP-doc of `set_date_start()` and `set_date_end()` as well.

Reply:
Done